src/main/java/com/zufar/icedlatte/payment/endpoint/PaymentEndpoint.java

```diff
@@ -29,117 +22,54 @@
 
+    // stripe listen --forward-to localhost:80/backend/api/v1/payment/stripe/webhook
     public static final String PAYMENT_URL = "/api/v1/payment";
 
     private final PaymentRetriever paymentRetriever;
     private final PaymentProcessor paymentProcessor;
     private final PaymentEventProcessor paymentEventProcessor;
-    private final CardDetailsProcessor cardDetailsProcessor;
-    private final SecurityPrincipalProvider securityPrincipalProvider;
-    private final ShoppingCartManager shoppingCartManager;
-    private final SingleUserProvider singleUserProvider;
-    private final StripeShippingOptionsProvider stripeShippingOptionsProvider;
-    private final StripeLineItemsConverter stripeLineItemsConverter;
-
-    @Value("${stripe.secret-key}")
-    public String secretKey;
+    private final StripeSessionProvider stripeSessionProvider;
 
     // TEST CARD: 4242424242424242
     @GetMapping
-    public ResponseEntity<PaymentSession> processPaymentTest() {
-        var userId = securityPrincipalProvider.getUserId();
-        log.info("Received request for processing payment for user {}", userId);
-        // TODO: create order
-        var user = singleUserProvider.getUserById(userId);
-        var shoppingCart = shoppingCartManager.getShoppingCartByUserId(userId);
-        log.info("For user {}", userId);
-        Stripe.apiKey = secretKey;
-        String DOMAIN = "http://localhost:80"; // FIXME: get this from request
-        SessionCreateParams params =
-                SessionCreateParams.builder()
-                        .setMode(SessionCreateParams.Mode.PAYMENT)
-                        .setUiMode(SessionCreateParams.UiMode.EMBEDDED)
-                        .setCustomerEmail(user.getEmail())
-                        .setReturnUrl(DOMAIN + "/orders?sessionId={CHECKOUT_SESSION_ID}")
-                        .addAllLineItem(stripeLineItemsConverter.getLineItems(shoppingCart))
-                        .addAllShippingOption(stripeShippingOptionsProvider.getShippingOptions())
-                        .setAutomaticTax(
-                                SessionCreateParams.AutomaticTax.builder()
-                                        .setEnabled(true)
-                                        .build()
-                        )
-                        .build();
-        Session session;
-        try {
-            session = Session.create(params);
-        } catch (StripeException e) {
-            throw new RuntimeException(e); // FIXME: create specific exception
-        } finally {
-            // FIXME: add record to the DB
-        }
-        shoppingCartManager.deleteByUserId(userId);
-        return ResponseEntity.status(HttpStatus.OK).body(new PaymentSession(session.getClientSecret()));
+    public ResponseEntity<SessionWithClientSecretDto> processPayment(@RequestParam final UUID orderId,
+                                                                     final HttpServletRequest request) throws OrderAlreadyPaidException {
```

Comment:
Why do you use `HttpServletRequest class` instead of custom `ProcessPaymentRequest class`?

Reply:
I need to construct return URL using actual scheme which is `http` for localhost and `https` for production, refer to com.zufar.icedlatte.payment.api.session.StripeSessionProvider#getReturnUrl